src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1265,81 +1268,176 @@ public void ForceRun()
         #region save/load
 
         /// <summary>
+        /// Opens a Dynamo workspace from a path to a file on disk.
+        /// </summary>
+        /// <param name="filePath">Path to file</param>
+        /// <param name="forceManualExecutionMode">Set this to true to discard
+        /// execution mode specified in the file and set manual mode</param>
+        public void OpenFileFromPath(string filePath, bool forceManualExecutionMode = false)
+        {
+          bool failed = false;
+          try
+          {
+            OpenXmlFileFromPath(filePath, forceManualExecutionMode);
+          }
+          catch (Exception e)
+          {
+            failed = true;
+          }
+
+          if (!failed)
+            return;
+
+          OpenJsonFileFromPath(filePath, forceManualExecutionMode);
```

Comment:
The methods have been updated to push the try...catch sections down into the file type specific methods.